src/screens/ChannelListScreen/hooks/useFollowUser.ts

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import React, {useCallback} from 'react';
+import {useIsFocused} from '@react-navigation/core';
+
+import following from '../../../context/actions/following';
+import {ChannelList as ChannelListData} from '../../../../types/database/schema/ChannelList.types';
+import {Context} from '../../../context';
+import {getFollowing, setFollow, setUnFollow} from '../../../service/profile';
+
+const useFollowUser = () => {
+  const isFocused = useIsFocused();
+  const [followContext, followingDispatch] = (React.useContext(Context) as unknown as any)
+    .following;
+  const [profileContext] = (React.useContext(Context) as unknown as any).profile;
+  const {myProfile} = profileContext;
+
+  const initialFollowingDataRef = React.useRef([...(followContext?.users ?? [])]);
+
+  const updateFollowingData = async () => {
+    try {
+      const response = await getFollowing();
+      following.setFollowingUsers(response.data, followingDispatch);
+    } catch (e) {
+      if (__DEV__) {
+        console.log('error: ', e);
+      }
+      throw new Error(e);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Error handling in the `updateFollowingData` function could be improved. Currently, it just logs the error and rethrows it. It would be better to handle the error gracefully and provide feedback to the user if necessary.
```diff
  const updateFollowingData = async () => {
    try {
      const response = await getFollowing();
      following.setFollowingUsers(response.data, followingDispatch);
    } catch (e) {
      if (__DEV__) {
        console.log('error: ', e);
      }
-       throw new Error(e);
+       // Handle error appropriately here
    }
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->